spec.bs

```diff
@@ -0,0 +1,369 @@
+spec:webidl; type:dfn; text:resolve
+</pre>
+
+<section>
+  <h2 id="introduction-header">Introduction</h2>
+
+  In today's web, people’s interests are typically inferred based on observing what sites or pages they visit, which relies on tracking techniques like third-party cookies or less-transparent mechanisms like device fingerprinting. It would be better for privacy if interest-based advertising could be accomplished without needing to collect a particular individual’s browsing history.
+
+  This specification provides an API to enable ad-targeting based on the people’s general browsing interest, without exposing the exact browsing history.
+
+  <div class="example">
+    Creating an ad based on the browsing interests, using the document.browsingTopics() JavaScript API:
+
+    (Inside a https://ads.example iframe)
+    <pre class="lang-js">
+      // document.browsingTopics() returns an array of BrowsingTopic objects.
+      const topics = await document.browsingTopics();
+
+      // Get data for an ad creative.
+      const response = await fetch('https://ads.example/get-creative', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(topics)
+      });
+
+      // Get the JSON from the response.
+      const creative = await response.json();
+
+      // Display the ad.
+    </pre>
+  </div>
+
+  <div class="example">
+    Creating an ad based on the browsing interests, based on the Sec-Browsing-Topics navigation request header:
+
+    (Inside the top level context)
+    <pre class="lang-js">
+      // A 'Sec-Browsing-Topics: [topics header value]' header will be sent in
+      // the HTTP request.
+      &ltiframe src="https://ads.example/get-creative"&gt&lt/iframe&gt
+    </pre>
+  </div>
+
+  <div class="example">
+    Creating an ad based on the browsing interests, based on the Sec-Browsing-Topics fetch() request header:
+
+    (Inside the top level context)
+    <pre class="lang-js">
+      // A 'Sec-Browsing-Topics: [topics header value]' header will be sent in
+      // the HTTP request.
+      const response = await fetch('https://ads.example/get-creative', {browsingTopics: true});
+      const ad_creative = await response.json();
+      // Display the ad.
+    </pre>
+  </div>
+</section>
+
+<section>
+  <h2 id="browsing-topic-dictionary-header">BrowsingTopic dictionary</h2>
+  The {{BrowsingTopic}} dictionary is used to contain the IDL correspondences of [=topic id=], [=version=], [=configuration version=], [=taxonomy version=], and [=model version=].
+
+  <pre class="idl">
+  dictionary BrowsingTopic {
+    long topic;
+    DOMString version;
+    DOMString configVersion;
+    DOMString modelVersion;
+    DOMString taxonomyVersion;
+  };
+  </pre>
+
+  A <dfn>taxonomy</dfn> comprises a list of advertising <dfn>topics</dfn>. A [=taxonomy=] is identified by a <dfn>taxonomy version</dfn>, and a [=topic=] is identified by a <dfn>topic id</dfn>.
```

Comment:
Is it possible to be more specific about the structure of any of these strings? E.g. are the versions numeric and increasing?

Reply:
I think we can require the versions to be integers (in string format/type).

The term “increasing” seems ambiguous to me. Do you mean increase with the increase of browser versions? But regardless, I think things would be more flexible to not require this. e.g. the browser could experiment with multiple versions, and settle on one version at a later time.